components/Valuation/Graph.tsx

```diff
@@ -0,0 +1,125 @@
+import React, { useEffect, useState, useRef } from "react";
+import { Metaverse } from "../../lib/enums";
+import { createChart, UTCTimestamp } from "lightweight-charts";
+import { getValuationDailyData } from "../../lib/FirebaseUtilities";
+
+const Graph = ({ metaverse }: Metaverse) => {
+  const metaverseGraph = useRef<HTMLDivElement>(null);
+  const [values, setValues] = useState([]);
```

Comment:
values here is typed as a `never[]` because there is no type specification. This is gonna cause you trouble with typescript later on down the code. You should make an interface of the response you get from the API const `[values, setValues] = useState<ApiValue[]>([]);`

This might be annoying at first but it helps you and makes you more precise for types. You already know this.